test/e2e/gardener/shoot/update-to-ha.go

```diff
@@ -0,0 +1,76 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package shoot
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	v1beta1helper "github.com/gardener/gardener/pkg/apis/core/v1beta1/helper"
+	. "github.com/gardener/gardener/test/e2e/gardener"
+	. "github.com/gardener/gardener/test/e2e/gardener/shoot/internal"
+	"github.com/gardener/gardener/test/e2e/gardener/shoot/internal/highavailability"
+	"github.com/gardener/gardener/test/e2e/gardener/shoot/internal/inclusterclient"
+)
+
+var _ = Describe("Shoot Tests", Label("Shoot", "high-availability"), func() {
+	container := func(shootName string, failureToleranceType gardencorev1beta1.FailureToleranceType) {
+		test := func(s *ShootContext) {
+			s.Shoot.Spec.ControlPlane = nil
```

Comment:
Nit
```suggestion
			s.Shoot.Spec.ControlPlane.HighAvailability = nil
```

Accordingly, `ItShouldUpdateShootToHighAvailability` should also mutate the `HighAvailability` field only.

Reply:
`ControlPlane` is a pointer type - it might not always be set. I changed `ItShouldUpdateShootToHighAvailability`, but I prefer to keep this line (or do I miss something)?